### PR TITLE
Timespinner: rename some locations for clarity.

### PR DESCRIPTION
**Pull request by alwaysintreble**
Adjusted a lot of locations to be more clear on where they're located for when hints are received. Put bat jump chest and royal guard tiny room in logic when the player has talaria + wheel.

### CODE REVIEW

**Comment by Berserker66** [comment]
This'll be waiting for @JarnoWesthof approval.

**Comment by JarnoWesthof** [comment]
Id like some locations changed, and also need to fix some things as atm it will break client compatibility, i will go into bigger details later, but just putting this comment here so it hopefully wont be merged like this

File: worlds/timespinner/PyramidKeys.py [hunk 1]
```diff
@@ -6,7 +6,7 @@ def get_pyramid_keys_unlock(world: MultiWorld, player: int) -> str:
     present_teleportation_gates: Tuple[str, ...] = (
         "GateKittyBoss",
         "GateLeftLibrary",
-        "GateMilitairyGate",
+        "GateMilitaryGate",
```
Comment (JarnoWesthof) [inline worlds/timespinner/PyramidKeys.py:9]: We got to rename this back to GateMilitairyGate, as its part of contract between client and server, to fix this type we would first need to change the client to support both versions and then we can change the server side

File: worlds/timespinner/LogicMixin.py [hunk 2]
```diff
@@ -15,6 +15,9 @@ def _timespinner_has_forwarddash_doublejump(self, world: MultiWorld, player: int
     def _timespinner_has_doublejump_of_npc(self, world: MultiWorld, player: int) -> bool:
         return self._timespinner_has_upwarddash(world, player) or (self.has('Timespinner Wheel', player) and self._timespinner_has_doublejump(world, player))
 
+    def _timespinner_has_fastjump_on_npc(self, world: MultiWorld, player: int) -> bool:
+        return self.has('Timespinner Wheel', player) and self.has('Talaria Attachment', player)
```
Comment (JarnoWesthof) [inline worlds/timespinner/LogicMixin.py:19]: this should use self.has_all when checking two specific items

File: worlds/timespinner/Locations.py [hunk 3]
```diff
@@ -24,16 +24,16 @@ def get_locations(world: Optional[MultiWorld], player: Optional[int]) -> Tuple[L
         LocationData('Lower lake desolation', 'Not so secret room',  1337008, lambda state: state._timespinner_can_break_walls(world, player)),
         LocationData('Lower lake desolation', 'Tank chest',  1337009, lambda state: state._timespinner_has_timestop(world, player)),
         LocationData('Upper lake desolation', 'Oxygen recovery room',  1337010),
-        LocationData('Upper lake desolation', 'Lake secret',  1337011, lambda state: state._timespinner_can_break_walls(world, player)),
-        LocationData('Upper lake desolation', 'Double jump cave floor',  1337012, lambda state: state._timespinner_has_doublejump(world, player)),
-        LocationData('Upper lake desolation', 'Double jump cave platform',  1337013),
+        LocationData('Upper lake desolation', 'Upper desolated secret',  1337011, lambda state: state._timespinner_can_break_walls(world, player)),
```
Comment (JarnoWesthof) [inline worlds/timespinner/Locations.py:27]: The names are still a bit under discussion ill have a look at them later

**Review by JarnoWesthof — CHANGES_REQUESTED** [review]

File: worlds/timespinner/PyramidKeys.py (hunk 1 above)
Comment (alwaysintreble) [reply worlds/timespinner/PyramidKeys.py]: reverted

**Comment by JarnoWesthof** [comment]
There are some conflicts as i already let the logic changes be merged in 2.0 so they are already live

**Pull request closed without merging**